recipes/configs/phi3/mini_qlora_single_device.yaml

```diff
@@ -40,52 +42,50 @@ checkpointer:
   model_type: PHI3_MINI
 resume_from_checkpoint: False
 
-# Dataset and Sampler
+# Dataset
 dataset:
   _component_: torchtune.datasets.alpaca_cleaned_dataset
 seed: null
 shuffle: True
-batch_size: 2
 
-# Optimizer and Scheduler
+# Fine-tuning arguments
+epochs: 1
+max_steps_per_epoch: null
+batch_size: 2
+gradient_accumulation_steps: 16
 optimizer:
   _component_: torch.optim.AdamW
   weight_decay: 0.01
   lr: 3e-4
 lr_scheduler:
   _component_: torchtune.modules.get_cosine_schedule_with_warmup
   num_warmup_steps: 100
-
 loss:
   _component_: torch.nn.CrossEntropyLoss
-
-# Training
-epochs: 1
-max_steps_per_epoch: null
-gradient_accumulation_steps: 16
 compile: False
 
+# Training env
+device: cuda
+
+# Memory management
+enable_activation_checkpointing: True
+dtype: bf16
+
 # Logging
-output_dir: /tmp/qlora_finetune_output/
 metric_logger:
   _component_: torchtune.utils.metric_logging.DiskLogger
-  log_dir: ${output_dir}
+  log_dir: /tmp/Phi-3-mini-4k-instruct/logs
 log_every_n_steps: 1
 log_peak_memory_stats: False
 
-# Environment
-device: cuda
-dtype: bf16
-enable_activation_checkpointing: True
-
-# Show case the usage of pytorch profiler
+# Showcase the usage of PyTorch profiler
 # Set enabled to False as it's only needed for debugging training
 profiler:
```

Comment:
I would love an explicit rule-of-thumb for when we include profiler by default. 

Anyone? @kartikayk @ebsmothers